### PR TITLE
fix(templates): Specify `AndroidManifest` correctly so that design-time builds don't fail

### LINKED ISSUE #11641: Central Package Management does not work with Mobile project
### Current behavior

When Central package management is used, the mobile project does not compile due to:

```
Projects that use central package version management should not define the version on the PackageReference items but on the PackageVersion items:
 Microsoft.Windows.SDK.BuildTools;Uno.Core.Extensions.Logging.Singleton;Microsoft.WindowsAppSDK;Uno.WinUI;Microsoft.Extensions.Logging.Console.
```

Even though the `PackageReferences` definitely don't have a `Version` attribute.

### Expected behavior

Should compile

### How to reproduce it (as minimally and precisely as possible)

- Clone https://github.com/MartinZikmund/uno-11641-repro
- Try to build Mobile project

### Workaround

Not using Central Package Management

### Works on UWP/WinUI

None

### Environment

_No response_

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #11641

For non-design-time builds, AndroidManifest is correctly set from here:

https://github.com/unoplatform/uno/blob/560a44980945eb8343025832fe0ea23941c99435/build/uno.winui.single-project.targets#L58

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SolutionTemplate/UnoSolutionTemplate.WinUI.netcore/Mobile/UnoQuickStart.Mobile.csproj [hunk 1]
```diff
@@ -26,6 +26,7 @@
 		<ApplicationDisplayVersion>1.0</ApplicationDisplayVersion>
 		<ApplicationVersion>1</ApplicationVersion>
 
+		<AndroidManifest>Android\AndroidManifest.xml</AndroidManifest>
```
Comment (jeromelaban) [inline src/SolutionTemplate/UnoSolutionTemplate.WinUI.netcore/Mobile/UnoQuickStart.Mobile.csproj:29]: This is a symptom of another problem, generally caused by nuget packages not being restored properly.

Comment (Youssef1313) [reply src/SolutionTemplate/UnoSolutionTemplate.WinUI.netcore/Mobile/UnoQuickStart.Mobile.csproj:29]: @jeromelaban NuGet needs a passing design-time build in order to restore, and the error about AndroidManifest happens in design-time builds by a target that is run before anything from NuGet runs. So, AndroidManifest needs to be set early.

Comment (MartinZikmund) [reply src/SolutionTemplate/UnoSolutionTemplate.WinUI.netcore/Mobile/UnoQuickStart.Mobile.csproj:29]: I assume same adjustment will be needed in the new templates as well (in Uno Extensions)

**Comment by MartinZikmund** [comment]
@jeromelaban for reference for the reason for this change - https://github.com/NuGet/Home/issues/12480

**Comment by jeromelaban** [comment]
The fix here is a workaround for the fact that we're not using an SDK (which can be executed prior to any design-time build). SDKExtras uses the `sdk/XX.XX` syntax, but this has its own set of issues (such as freezing the IDE because of dispatcher-bound external downloads).

This is not an ideal solution, but as nuget relies on design-time builds to work, this is the only way to work around it at this time, indeed.

**Review by MartinZikmund — APPROVED** [review]

**Pull request merged**